### PR TITLE
[ADD][12.0] invoice_customer_ref

### PR DESCRIPTION
**Pull request by Daniel-CA**

### CODE REVIEW

**Comment by Daniel-CA** [comment]
@avanzosc/developers review please

**Review by alfredoavanzosc — APPROVED** [review]
:+1:

File: invoice_customer_ref/models/invoice.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    client_order_ref = fields.Char(string='Customer Reference', copy=False)
+
+    def write(self, vals):
+        if 'client_order_ref' in vals:
+            inv_line_obj = self.env['account.invoice.line']
+            for invoice in self:
+                if 'sale_loop' in self.env.context:
+                    del vals['client_order_ref']
+                    continue
+                sale_orders = inv_line_obj.search(
```
Comment (oihane) [inline invoice_customer_ref/models/invoice.py]: sale_orders = self.mapped('invoice_line_ids.sale_line_ids.order_id')

File: invoice_customer_ref/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    def write(self, vals):
+        if 'client_order_ref' in vals:
+            for sale in self:
+                if 'invoice_loop' in self.env.context:
+                    del vals['client_order_ref']
+                    continue
+                invoice_ids = sale.order_line.mapped(
```
Comment (oihane) [inline invoice_customer_ref/models/sale_order.py]: por si acaso:
invoices = self.mapped('order_line.invoice_lines.invoice_id').filtered....

**Review by oihane — APPROVED** [review]

**Pull request merged**